pywinauto/controls/uia_controls.py

```diff
@@ -134,22 +136,89 @@ def __init__(self, elem):
+            else:
+                try:
+                    self.invoke()
+                except NoPatternInterfaceError:
+                    raise NoPatternInterfaceError('There is no ExpandCollapsePattern and ' \
+                        'no "Open" button in .children(). Maybe only .click_input() would help to expand.')
+        return self
+
+    # -----------------------------------------------------------
+    def collapse(self):
+        if not self.is_expanded():
+            return self
+        try:
+            super(ComboBoxWrapper, self).collapse()
+        except NoPatternInterfaceError:
+            # workaround for WinForms combo box using Open button
+            close_buttons = self.children(title='Close', control_type='Button')
+            if not close_buttons:
+                if self.element_info.framework_id == 'WinForm':
+                    return self # simple WinForms combo box is always expanded
+                else:
+                    raise RuntimeError('There is no ExpandCollapsePattern and no "Close" button for the combo box')
+            if self.is_editable():
+                close_buttons[0].click_input()
+            else:
+                close_buttons[0].invoke()
+        return self
+
+    # -----------------------------------------------------------
+    def is_editable(self):
+        edit_children = self.children(control_type="Edit")
+        return len(edit_children) > 0
+
+    # -----------------------------------------------------------
+    def get_expand_state(self):
+        try:
+            return super(ComboBoxWrapper, self).get_expand_state()
+        except NoPatternInterfaceError:
+            # workaround for WinForms combo box
+            children_list = self.children(control_type="List")
+            if children_list and children_list[0].is_visible():
+                return uia_defs.expand_state_expanded
+            else:
+                return uia_defs.expand_state_collapsed
+
     # -----------------------------------------------------------
     def texts(self):
         """Return the text of the items in the combobox"""
         texts = []
         # ComboBox has to be expanded to populate a list of its children items
         try:
-            self.expand()
+            super(ComboBoxWrapper, self).expand()
             for c in self.children():
                 texts.append(c.window_text())
         except NoPatternInterfaceError:
-            return texts
+            children_lists = self.children(control_type='List')
+            if children_lists:
+                # workaround for Qt5 and WinForms
+                return children_lists[0].children_texts()
+            elif self.handle:
+                # workaround using "win32" backend
+                win32_combo = win32_controls.ComboBoxWrapper(self.handle)
+                texts.extend(win32_combo.item_texts())
+            else:
+                # TODO: maybe add more workarounds
+                return texts
```

Comment:
Maybe worth removing this clause to improve the coverage. BTW, I wonder if it's possible to not have self.handle ?

Reply:
Maybe some custom WPF combo box. I'm not willing to ignore this case. It will just return an empty list.